schema/generate_scaler_schema.go

```diff
@@ -0,0 +1,550 @@
+				if !ok {
+					continue
+				}
+
+				r, ok := c.Body[0].(*ast.ReturnStmt)
+				if !ok {
+					continue
+				}
+
+				if len(r.Results) == 0 {
+					continue
+				}
+				caller, ok := r.Results[0].(*ast.CallExpr)
+				if !ok {
+					continue
+				}
+
+				expr, ok := caller.Fun.(*ast.SelectorExpr)
+				if !ok {
+					continue
+				}
+
+				scalerCallers[strings.Trim(basic.Value, "\"")] = expr.Sel.Name
+			}
+		}
+		return true
+	})
+	return scalerCallers, nil
+}
+
+// getAllKedaTagedStructs is a function that gets all the structs that are tagged with `keda` from the scalers files
+func getAllKedaTagedStructs(dir string) (map[string]*ast.StructType, map[string]*ast.StructType) {
+	// loop all files in the scalers directory and get the structs that are tagged with `keda`
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	kedaScalerStructs := map[string]*ast.StructType{}
+	kedaTagStructs := map[string]*ast.StructType{}
+
+	for _, e := range entries {
+		if e.IsDir() {
+			getAllKedaTagedStructs(dir + "/" + e.Name())
+			continue
+		}
+
+		data, err := os.ReadFile(dir + "/" + e.Name())
+		if err != nil {
+			continue
+		}
+
+		fset := token.NewFileSet()
+		f, err := parser.ParseFile(fset, "", string(data), 0)
+		if err != nil {
+			continue
+		}
+
+		var scalerStructs *ast.StructType
+
+		for _, decl := range f.Decls {
+			switch v := decl.(type) {
+			case *ast.GenDecl:
+				for _, spec := range v.Specs {
+					ts, ok := spec.(*ast.TypeSpec)
+					if !ok {
+						continue
+					}
+
+					structType, ok := ts.Type.(*ast.StructType)
+					if !ok {
+						continue
+					}
+
+					hasTriggerIndex := false
+					for _, v := range structType.Fields.List {
+						// Find `triggerIndex` field and identify the struct as a scaler struct, otherwise identify the struct as an abstract substruct
+						if len(v.Names) > 0 && v.Names[0].Name == "triggerIndex" {
+							hasTriggerIndex = true
+							continue
+						}
+
+						if v.Tag == nil || v.Tag.Value == "" {
+							continue
+						}
+
+						tagvalue := strings.Split(strings.Trim(v.Tag.Value, "`"), ":")
+
+						// if the tag is `keda`, add the struct to the kedaTagStructs map
+						if len(tagvalue) > 0 && tagvalue[0] == "keda" {
+							kedaTagStructs[ts.Name.Name] = structType
+						}
+					}
+
+					if hasTriggerIndex {
+						scalerStructs = kedaTagStructs[ts.Name.Name]
+						delete(kedaTagStructs, ts.Name.Name)
+					}
+				}
+			case *ast.FuncDecl:
+				if strings.HasPrefix(v.Name.Name, creatorSymbol) && scalerStructs != nil {
+					kedaScalerStructs[v.Name.Name] = scalerStructs
+				}
+			}
+		}
+	}
+
+	return kedaScalerStructs, kedaTagStructs
+}
+
+func main() {
+	var builderFilePath string
+	var scalersFilesDirPath string
+	var specifyScaler string
+	var outputFileName string
+	var outputFilePath string
+	var outputFormat string
+	pflag.StringVar(&kedaVersion, "keda-version", "1.0", "Set the version of current KEDA in schema.")
```

Comment:
nit: maybe default to `main`? KEDA version `1.0` might be slightly confusing with
https://github.com/kedacore/keda/releases/tag/v1.0.0